src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -1904,46 +1975,89 @@ public void addRoute(String remoteId, String uuid, int metric) {
     routeTable.addRoute(remoteId, uuid, metric);
   }
 
-  static public ServiceInterface start(String name) {
-    ServiceInterface si = null;
-    try {
-      si = create(name);
-    } catch (Exception e) {
-      Runtime.getInstance().error(e);
-      log.error("create({}) threw", name, e);
-    }
+  static public ServiceInterface start() {
+    return start(null, null, null);
+  }
 
-    if (si != null) {
-      return start(name, si.getType());
-    }
+  static public ServiceInterface start(String name, String type) {
+    return start(null, name, type);
+  }
 
-    return null;
+  static public ServiceInterface start(String name) {
+    return start(null, name, null);
   }
 
-  static public ServiceInterface start(String name, String type) {
+  static public ServiceInterface start(Plan inPlan, String name) {
+    return start(inPlan, name, null);
+  }
+
+  /**
+   * start(name, type) goes through the full service lifecycle of:
+   * 
+   * <pre>
+   * clear  - clearing the plan for construction of service(s) needed 
+   * load   - loading the plan for desired services 
+   * check  - checking all planned service have met appropriate licensing and dependency checks create -
+   * create - create the service and its peers
+   * start  - start the service and all specified
+   * </pre>
+   * 
+   * if name and type are null - then process whatever is currently in the Plan
+   * 
+   * @param name
+   * @param type
+   *          - can be null, if file exists on the file system or already loaded
+   * @return
+   */
+  static public ServiceInterface start(Plan inPlan, String name, String type) {
+    // hand back immediately if a service with that name exists
+    // and is running
+    if (inPlan == null && runtime != null) {
+      inPlan = runtime.plan;
+    }
+    
     ServiceInterface si = null;
-    // framework level catch of all startServices
-    // we will catch it here and log it with a stack trace
-    // its not a good idea to let exceptions propegate higher - because
-    // logging format can get challenging (Python trace-back) or they may
-    // be completely lost - this is the last level the error can be handled
-    // before
-    // going into the unknown - so we catch it !
-    try {
-      si = create(name, type);
-      if (si != null) {
+
+    si = Runtime.getService(name);
+
+    if (si != null) {
+      if (si.isRunning()) {
+        return si;
+      } else {
         si.startService();
+        return si;
       }
+    }
 
-    } catch (Exception e) {
-      String error = String.format("createAndStart(%s, %s) %s", name, type, e.getClass().getCanonicalName());
-      Runtime.getInstance().error(error);
-      log.error(error, e);
+    si = create(inPlan, name, type);
+    if (si != null && !si.isRunning()) {
+      si.startService();
     }
 
+    // clear the plan
+    Runtime.clear();
+
     return si;
   }
 
+  public static ServiceConfig load(String name, String type) {
+    return load(null, name, type);

```

Comment:
don't pass null as the first arg.. that's confusing.. if you're not passing a plan here. .just call load(name,type)

Reply:
this commit - peers needed to send in their plan
removing this ..
there is only "one" plan